core/notification/src/main/java/edu/stanford/spezi/core/notification/setting/NotificationSettingsStateMapper.kt

```diff
@@ -0,0 +1,16 @@
+package edu.stanford.spezi.core.notification.setting
+
+import javax.inject.Inject
+
+/**
+ * Maps [NotificationSettingViewModel.Action] to [NotificationSettingViewModel.UiState].
+ */
+internal class NotificationSettingsStateMapper @Inject constructor() {
+    fun mapSwitchChanged(
+        action: NotificationSettingViewModel.Action.SwitchChanged,
+        currentSettings: NotificationSettings,
+    ): NotificationSettings {
+        val updatedSettings = currentSettings.update(action.notificationType, action.isChecked)
+        return updatedSettings.copy(pendingActions = currentSettings.pendingActions + action.notificationType)
```

Comment:
As proposed in the other comment, I would move the pending actions as a property of the ui state. Also here this operation is not being used as the mapped settings are just used to be saved in firebase. you are again updating the pending actions in the view model on current settings